dask_cloudprovider/azure/azurevm.py

```diff
@@ -489,21 +502,21 @@ def __init__(
         self.network_client = NetworkManagementClient(
             self.credentials, self.subscription_id
         )
-        self.vnet = vnet if vnet is not None else self.config.get("vnet")
+        self.vnet = vnet if vnet is not None else self.config.get("azurevm.vnet")
         if self.vnet is None:
             raise ConfigError("You must configure a vnet")
         self.security_group = (
             security_group
             if security_group is not None
-            else self.config.get("security_group")
+            else self.config.get("azurevm.security_group")
         )
         if self.security_group is None:
             raise ConfigError(
                 "You must configure a security group which allows traffic on 8786 and 8787"
             )
-        self.vm_size = vm_size if vm_size is not None else self.config.get("vm_size")
+        self.vm_size = vm_size if vm_size is not None else self.config.get("azurevm.vm_size")
```

Comment:
While you're here we could simplify each of these using the `override_with` kwarg that you've helpfully exposed in the custom config object.

```suggestion
        self.vm_size = self.config.get("azurevm.vm_size", override_with=vm_size)
```

Reply:
Oh yeah 🤦